pandas/core/internals/blocks.py

```diff
@@ -2616,36 +2623,38 @@ def get_block_type(values, dtype=None):
     -------
     cls : class, subclass of Block
     """
+    # We use vtype and kind checks because they are much more performant
+    #  than is_foo_dtype
     dtype = dtype or values.dtype
     vtype = dtype.type
+    kind = dtype.kind
 
     cls: Type[Block]
 
     if is_sparse(dtype):
         # Need this first(ish) so that Sparse[datetime] is sparse
         cls = ExtensionBlock
-    elif is_categorical_dtype(values.dtype):
+    elif dtype.name == "category":
```

Comment:
why not `is Categorical` ? e.g. since we are removing comparison vs 'category' generally (in your other PR)

Reply:
could do `isinstance(dtype, CategoricalDtype)`.  either way is fine by me

Reply:
i think prefer that

Reply:
updated+green